### PR TITLE
Port netstring and hashutil to python 3

### PR DESCRIPTION
**Pull request by itamarst**
Fixes ticket:3344

This builds on #744 so that should be reviewed and merged first.

### CODE REVIEW

File: src/allmydata/test/test_hashutil.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        self._testknown(hashutil.ssk_pubkey_fingerprint_hash, b"3opzw4hhm2sgncjx224qmt5ipqgagn7h5zivnfzqycvgqgmgz35q", b"")
+        self._testknown(hashutil.ssk_readkey_hash, b"vugid4as6qbqgeq2xczvvcedai", b"")
+        self._testknown(hashutil.ssk_readkey_data_hash, b"73wsaldnvdzqaf7v4pzbr2ae5a", b"iv", b"rk")
+        self._testknown(hashutil.ssk_storage_index_hash, b"j7icz6kigb6hxrej3tv4z7ayym", b"")
+
+        self._testknown(hashutil.permute_server_hash,
+                        b"kb4354zeeurpo3ze5e275wzbynm6hlap", # b32(expected)
+                        b"SI", # peer selection index == storage_index
+                        base32.a2b(b"u33m4y7klhz3bypswqkozwetvabelhxt"), # seed
+                        )
```
Comment (exarkun) [inline src/allmydata/test/test_hashutil.py:135]: I assume all of this code is moved from `test_util.py` with only substitution of byte string literals for native string literals and no other changes.

**Comment by exarkun** [comment]
Ugh.  I just noticed there are only *10* green checks on this PR.  Where did ... all of CircleCI go? :( :( :(

**Comment by exarkun** [comment]
> Resolved - From 13:35 to 13:39 UTC, there may have been potential delays or failure of jobs starting. This was due to a database failover. As of 13:39 UTC, our systems have fully recovered and all jobs are starting successfully.

sigh.  so maybe it is going to start eventually?

File: src/allmydata/test/test_hashutil.py (hunk 1 above)
Comment (itamarst) [reply src/allmydata/test/test_hashutil.py:135]: Yeah.

**Pull request merged**